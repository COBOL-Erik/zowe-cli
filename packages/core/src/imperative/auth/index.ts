/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

export * from "./builders/AuthCommandBuilder";
export * from "./builders/AuthLoginCommandBuilder";
export * from "./builders/AuthLogoutCommandBuilder";
export * from "./builders/CompleteAuthGroupBuilder";

export * from "./doc/IAuthHandlerApi";

export * from "./handlers/AbstractAuthHandler";
export * from "./handlers/BaseAuthHandler";
