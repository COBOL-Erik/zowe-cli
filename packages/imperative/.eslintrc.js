module.exports = {
    ...require("../../.eslintrc.js"),
    "parserOptions": {
        "ecmaVersion": 12,
        "sourceType": "module",
        "project": [
            "./tsconfig.json",
            "./__tests__/tsconfig.json",
            "./tsconfig-tests.json"
        ]
    }
}