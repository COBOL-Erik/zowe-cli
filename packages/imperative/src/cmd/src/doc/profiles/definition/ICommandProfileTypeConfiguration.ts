/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { ICommandProfileSchema } from "./ICommandProfileSchema";
import { IProfileTypeConfiguration } from "../../../../..";
import { ICommandProfileAuthConfig } from "./ICommandProfileAuthConfig";

/**
 * Cmd packages additions to the profile manager type configuration document. Used by the CliProfileManager. Allows
 * profiles to be built from command arguments. See the "CliProfileManager" for more information.
 * @export
 * @interface ICommandProfileTypeConfiguration
 * @extends {IProfileTypeConfiguration}
 */
export interface ICommandProfileTypeConfiguration extends IProfileTypeConfiguration {
    /**
     * The JSON schema document. The schema document provides a way to enforce the contents of a profile. The schema
     * conforms exactly to the JSON schema specification. You must supply all properties you would like validated
     * for correctness on the schema, except for "dependencies". Dependency schema checking is generated automatically
     * if you populate the dependencies property of this document.
     *
     * @type {ICommandProfileSchema}
     */
    schema: ICommandProfileSchema;

    /**
     * Configuration for authentication services to associate with this profile type.
     */
    authConfig?: ICommandProfileAuthConfig[];
}
