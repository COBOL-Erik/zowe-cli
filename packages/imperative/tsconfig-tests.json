{
  "compilerOptions": {
    "types": [
      "node",
      "jest"
    ],
    "lib": [
      "esnext"
    ],
    "experimentalDecorators": true,
    "target": "es2015",
    "module": "commonjs",
    "noEmit": true,
    "declaration": true,
    "moduleResolution": "node",
    "noImplicitAny": true,
    "outDir": "./lib",
    "preserveConstEnums": true,
    "removeComments": false,
    "pretty": true,
    "sourceMap": true,
    "newLine": "lf"
  },
  "typeRoots": [
    "./node_modules/@types"
  ],
  "include": [
    "**/__tests__/*",
    "**/__tests__/**/*",
    "**/__mocks__/*"
  ],
  "exclude": [
    "lib",
    "node_modules"
  ],
  "files": [
    "../../__types__/wontache.d.ts"
  ],
  "allowJs": true
}
