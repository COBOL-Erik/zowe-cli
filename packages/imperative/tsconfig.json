{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "./lib",
    "stripInternal": true
  },
  "include": [
    "./src"
  ]
}