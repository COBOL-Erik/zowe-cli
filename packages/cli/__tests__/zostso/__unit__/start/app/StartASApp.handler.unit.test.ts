/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

import { ZosmfRestClient } from "@zowe/core-for-zowe-sdk";
import { StartTso, AddressSpaceApps, IStartStopResponses } from "@zowe/zos-tso-for-zowe-sdk";
import * as StartASAppHandler from "../../../../../src/zostso/start/as-app/StartASApp.handler";
import * as StartASAppDefinition from "../../../../../src/zostso/start/as-app/StartASApp.definition";
import { IHandlerParameters } from "@zowe/imperative";
import { mockHandlerParameters } from "@zowe/cli-test-utils";
import { UNIT_TEST_ZOSMF_PROF_OPTS } from "../../../../../../../__tests__/__src__/TestConstants";

const NEW_AS_SPACE_PARAMS: IHandlerParameters = mockHandlerParameters({
    arguments: UNIT_TEST_ZOSMF_PROF_OPTS,
    positionals: ["zos-tso", "start", "app"],
    definition: StartASAppDefinition.StartASApp,
});
const EXISTING_AS_SPACE_PARAMS: IHandlerParameters = mockHandlerParameters({
    arguments: UNIT_TEST_ZOSMF_PROF_OPTS,
    positionals: ["zos-tso", "start", "app"],
    definition: StartASAppDefinition.StartASApp,
});
const MOCK_RESPONSE = Promise.resolve({
    version: "0100",
    reused: false,
    timeout: false,
    servletKey: "ZOWEUSER-123-aaaaaa",
    queueID: "983068",
    tsoData: [
        {
            "TSO MESSAGE": {
                VERSION: "0100",
                DATA: "HELLOW exec processing has started.",
            },
        },
        {
            "TSO MESSAGE": {
                VERSION: "0100",
                DATA: "UNIX message queue id = 983068",
            },
        },
        {
            "TSO MESSAGE": {
                VERSION: "0100",
                DATA: "Input message type = 32772",
            },
        },
        { "TSO MESSAGE": { VERSION: "0100", DATA: "Output message type = 4" } },
        {
            "TSO MESSAGE": {
                VERSION: "0100",
                DATA: "Reading application input from the UNIX message queue.",
            },
        },
    ],
});

const MOCK_START_RESPONSE: Promise<IStartStopResponses> = Promise.resolve({
    collectedResponses: [],
    messages:
        "IKJ56455I ZOWEUSER LOGON IN PROGRESS AT 11:18:56 ON OCTOBER 14, 2024\nIKJ56951I NO BROADCAST MESSAGES\nREADY \n",
    servletKey: "ZOWEUSER-123-aaaaaa",
    success: true,
    zosmfTsoResponse: {
        ver: "0100",
        queueID: "983068",
        reused: false,
        servletKey: "ZOWEUSER-123-aaaaaa",
        sessionID: "0x00",
        timeout: false,
        tsoData: [{}],
    },
});
describe("receive TSO app handler behavior", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("should properly start TSO address space and run an application at the created address space", async () => {
        jest.spyOn(StartTso, "start").mockReturnValue(MOCK_START_RESPONSE);
        jest.spyOn(ZosmfRestClient, "postExpectJSON").mockReturnValue(
            MOCK_RESPONSE
        );

        const startSpy = jest.spyOn(AddressSpaceApps, "start");
        const startASSpaceSpy = jest.spyOn(StartTso, "start");
        const zosmfSpy = jest.spyOn(ZosmfRestClient, "postExpectJSON");
        const handler = new StartASAppHandler.default();
        const params = Object.assign({}, ...[NEW_AS_SPACE_PARAMS]);
        let error = undefined;
        params.arguments = {
            ...params.arguments,
            account: "izuacct",
            startup: "EXEC 'ZOWEUSER.PUBLIC.REXX(VAREXX)'",
            appKey: "test2",
        };
        try {
            await handler.process(params);
        } catch (e) {
            error = e;
        }
        expect(error).toBeUndefined();
        expect(startSpy).toHaveBeenCalledTimes(1);
        expect(startASSpaceSpy).toHaveBeenCalledTimes(1);
        expect(zosmfSpy).toHaveBeenCalledTimes(1);
    });
    it("should properly start TSO address space at an existing TSO address space", async () => {
        jest.spyOn(StartTso, "start").mockReturnValue(MOCK_START_RESPONSE);
        jest.spyOn(ZosmfRestClient, "postExpectJSON").mockReturnValue(
            MOCK_RESPONSE
        );

        const startSpy = jest.spyOn(AddressSpaceApps, "start");
        const startASSpaceSpy = jest.spyOn(StartTso, "start");
        const zosmfSpy = jest.spyOn(ZosmfRestClient, "postExpectJSON");
        const handler = new StartASAppHandler.default();
        const params = Object.assign({}, ...[EXISTING_AS_SPACE_PARAMS]);
        let error = undefined;
        params.arguments = {
            ...params.arguments,
            account: "izuacct",
            startup: "EXEC 'ZOWEUSER.PUBLIC.REXX(VAREXX)'",
            queueId: "983068",
            servletKey: "ZOWEUSER-123-aaaaaa",
            appKey: "test2",
        };
        try {
            await handler.process(params);
        } catch (e) {
            error = e;
        }
        expect(error).toBeUndefined();
        expect(startSpy).toHaveBeenCalledTimes(1);
        expect(startASSpaceSpy).toHaveBeenCalledTimes(0);
        expect(zosmfSpy).toHaveBeenCalledTimes(1);
    });
});
