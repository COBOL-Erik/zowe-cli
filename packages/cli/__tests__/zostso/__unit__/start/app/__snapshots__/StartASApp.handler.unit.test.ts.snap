// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`receive TSO app handler behavior should properly start TSO address space and run an application at the created address space 1`] = `
"{
  \\"reused\\": false,
  \\"timeout\\": false,
  \\"servletKey\\": \\"CUST009-123-aaaaaa\\",
  \\"queueID\\": \\"983068\\",
  \\"tsoData\\": [
    {
      \\"VERSION\\": \\"0100\\",
      \\"DATA\\": \\"HELLOW exec processing has started.\\"
    },
    {
      \\"VERSION\\": \\"0100\\",
      \\"DATA\\": \\"UNIX message queue id = 983068\\"
    },
    {
      \\"VERSION\\": \\"0100\\",
      \\"DATA\\": \\"Input message type = 32772\\"
    },
    {
      \\"VERSION\\": \\"0100\\",
      \\"DATA\\": \\"Output message type = 4\\"
    },
    {
      \\"VERSION\\": \\"0100\\",
      \\"DATA\\": \\"Reading application input from the UNIX message queue.\\"
    }
  ]
}"
`;

exports[`receive TSO app handler behavior should properly start TSO address space at an existing TSO address space 1`] = `
"{
  \\"reused\\": false,
  \\"timeout\\": false,
  \\"servletKey\\": \\"CUST009-123-aaaaaa\\",
  \\"queueID\\": \\"983068\\",
  \\"tsoData\\": [
    {
      \\"VERSION\\": \\"0100\\",
      \\"DATA\\": \\"HELLOW exec processing has started.\\"
    },
    {
      \\"VERSION\\": \\"0100\\",
      \\"DATA\\": \\"UNIX message queue id = 983068\\"
    },
    {
      \\"VERSION\\": \\"0100\\",
      \\"DATA\\": \\"Input message type = 32772\\"
    },
    {
      \\"VERSION\\": \\"0100\\",
      \\"DATA\\": \\"Output message type = 4\\"
    },
    {
      \\"VERSION\\": \\"0100\\",
      \\"DATA\\": \\"Reading application input from the UNIX message queue.\\"
    }
  ]
}"
`;
